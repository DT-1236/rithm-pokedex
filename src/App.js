import React, { Component } from 'react';
import Pokedex from './Pokedex';
import './App.css';

class App extends Component {
  static defaultProps = {
    pokemon: [
      { id: 4, name: 'Charmander', type: 'fire', base_experience: 62 },
      { id: 7, name: 'Squirtle', type: 'water', base_experience: 63 },
      { id: 11, name: 'Metapod', type: 'bug', base_experience: 72 },
      { id: 12, name: 'Butterfree', type: 'flying', base_experience: 178 },
      { id: 25, name: 'Pikachu', type: 'electric', base_experience: 112 },
      { id: 39, name: 'Jigglypuff', type: 'normal', base_experience: 95 },
      { id: 94, name: 'Gengar', type: 'poison', base_experience: 225 },
      { id: 133, name: 'Eevee', type: 'normal', base_experience: 65 }
    ]
  };

  render() {
    // get random pokemon and assign to two decks
    const [deck1, deck2] = getPokemon(this.props.pokemon);

    // calculate base experience for deck1
    const exp1 = deck1.reduce((tally, p) => {
      tally += p.base_experience;
      return tally;
    }, 0);

    // calculate base experience for deck1
    const exp2 = deck2.reduce((tally, p) => {
      tally += p.base_experience;
      return tally;
    }, 0);

    console.log(deck1, deck2, exp1, exp2);

    return (
      <div className="App">
        <Pokedex pokemon={deck1} isWinner={exp1 > exp2} />
        <Pokedex pokemon={deck2} isWinner={exp2 > exp1} />
      </div>
    );
  }
}

const getPokemon = pokemon => {
  // Fisher-Yates shuffle algorithm
  for (let i = pokemon.length - 1; i > 0; i -= 1) {
    let randomIndex = Math.floor(Math.random() * i);
    let placeholder = pokemon[i];
    pokemon[i] = pokemon[randomIndex];
    pokemon[randomIndex] = placeholder;
  }

  const deck1 = pokemon.slice(0, 4);
  const deck2 = pokemon.slice(4);

  return [deck1, deck2];
};

export default App;
