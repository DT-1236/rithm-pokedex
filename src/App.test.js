import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

it('renders without crashing', () => {
  shallow(<App />);
});

it('renders using mount', () => {
  mount(<App />);
});

it('matches snapshot', () => {
  let wrapper = mount(<App />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
