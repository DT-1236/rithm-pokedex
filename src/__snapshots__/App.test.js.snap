// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <Pokedex
      pokemon={
        Array [
          Object {
            "base_experience": 62,
            "id": 4,
            "name": "Charmander",
            "type": "fire",
          },
          Object {
            "base_experience": 63,
            "id": 7,
            "name": "Squirtle",
            "type": "water",
          },
          Object {
            "base_experience": 72,
            "id": 11,
            "name": "Metapod",
            "type": "bug",
          },
          Object {
            "base_experience": 178,
            "id": 12,
            "name": "Butterfree",
            "type": "flying",
          },
          Object {
            "base_experience": 112,
            "id": 25,
            "name": "Pikachu",
            "type": "electric",
          },
          Object {
            "base_experience": 95,
            "id": 39,
            "name": "Jigglypuff",
            "type": "normal",
          },
          Object {
            "base_experience": 225,
            "id": 94,
            "name": "Gengar",
            "type": "poison",
          },
          Object {
            "base_experience": 65,
            "id": 133,
            "name": "Eevee",
            "type": "normal",
          },
        ]
      }
    >
      <div
        className="d-flex flex-wrap"
      >
        <Pokecard
          experience={62}
          id={4}
          name="Charmander"
          type="fire"
        >
          <div
            className="card"
          >
            <h3>
              Charmander
            </h3>
            <img
              alt=""
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
            />
            <p>
              Type: 
              fire
            </p>
            <p>
              EXP: 
              62
            </p>
          </div>
        </Pokecard>
        <Pokecard
          experience={63}
          id={7}
          name="Squirtle"
          type="water"
        >
          <div
            className="card"
          >
            <h3>
              Squirtle
            </h3>
            <img
              alt=""
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/7.png"
            />
            <p>
              Type: 
              water
            </p>
            <p>
              EXP: 
              63
            </p>
          </div>
        </Pokecard>
        <Pokecard
          experience={72}
          id={11}
          name="Metapod"
          type="bug"
        >
          <div
            className="card"
          >
            <h3>
              Metapod
            </h3>
            <img
              alt=""
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/11.png"
            />
            <p>
              Type: 
              bug
            </p>
            <p>
              EXP: 
              72
            </p>
          </div>
        </Pokecard>
        <Pokecard
          experience={178}
          id={12}
          name="Butterfree"
          type="flying"
        >
          <div
            className="card"
          >
            <h3>
              Butterfree
            </h3>
            <img
              alt=""
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/12.png"
            />
            <p>
              Type: 
              flying
            </p>
            <p>
              EXP: 
              178
            </p>
          </div>
        </Pokecard>
        <Pokecard
          experience={112}
          id={25}
          name="Pikachu"
          type="electric"
        >
          <div
            className="card"
          >
            <h3>
              Pikachu
            </h3>
            <img
              alt=""
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"
            />
            <p>
              Type: 
              electric
            </p>
            <p>
              EXP: 
              112
            </p>
          </div>
        </Pokecard>
        <Pokecard
          experience={95}
          id={39}
          name="Jigglypuff"
          type="normal"
        >
          <div
            className="card"
          >
            <h3>
              Jigglypuff
            </h3>
            <img
              alt=""
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/39.png"
            />
            <p>
              Type: 
              normal
            </p>
            <p>
              EXP: 
              95
            </p>
          </div>
        </Pokecard>
        <Pokecard
          experience={225}
          id={94}
          name="Gengar"
          type="poison"
        >
          <div
            className="card"
          >
            <h3>
              Gengar
            </h3>
            <img
              alt=""
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/94.png"
            />
            <p>
              Type: 
              poison
            </p>
            <p>
              EXP: 
              225
            </p>
          </div>
        </Pokecard>
        <Pokecard
          experience={65}
          id={133}
          name="Eevee"
          type="normal"
        >
          <div
            className="card"
          >
            <h3>
              Eevee
            </h3>
            <img
              alt=""
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/133.png"
            />
            <p>
              Type: 
              normal
            </p>
            <p>
              EXP: 
              65
            </p>
          </div>
        </Pokecard>
      </div>
    </Pokedex>
  </div>
</App>
`;
